proctor-common/src/main/java/com/indeed/proctor/common/model/TestType.java

```diff
@@ -34,6 +39,18 @@ public static TestType register(@Nonnull final String name) {
         return previous != null ? previous : testType;
     }
 
+    public void addDependency(@Nonnull final TestType dependency) {
+        allowedDependencies.add(dependency.name);
+    }
+
+    public boolean isValidDependency(@Nonnull final TestType dependency) {
+        return allowedDependencies.contains(dependency.name);
+    }
+
+    public String dependenciesToString() {
+        return String.join(", ", allowedDependencies);
```

Comment:
The ordering of `allowedDependencies` is not guaranteed because "It makes no guarantees as to the iteration order of the set; in particular, it does not guarantee that the order will remain constant over time" (https://docs.oracle.com/javase/8/docs/api/java/util/HashSet.html). You might want to sort this (e.g., `return allowedDependencies.stream().sorted().collect(Collectors.joining(", "))`) to make it consistent so that unit tests can pass always.